bpf/tests/lib/egressgw.h

```diff
@@ -271,3 +279,224 @@ static __always_inline int create_ct_entry(struct __ctx_buff *ctx, __be16 client
 	return ct_create4(get_ct_map4(&tuple), &CT_MAP_ANY4, &tuple, ctx,
 			 CT_EGRESS, &ct_state, NULL);
 }
+
+#ifdef ENABLE_IPV6
+static __always_inline int egressgw_pktgen_v6(struct __ctx_buff *ctx,
+					      struct egressgw_test_ctx test_ctx)
+{
+	union v6addr ext_svc_ip = EXTERNAL_SVC_IP_V6;
+	union v6addr client_ip = CLIENT_IP_V6;
+	union v6addr egress_ip = EGRESS_IP_V6;
+	union v6addr egress_ip3 = EGRESS_IP3_V6;
+	struct pktgen builder;
+	struct tcphdr *l4;
+	struct ethhdr *l2;
+	struct ipv6hdr *l3;
+	void *data;
+
+	/* Init packet builder */
+	pktgen__init(&builder, ctx);
+
+	/* Push ethernet header */
+	l2 = pktgen__push_ethhdr(&builder);
+	if (!l2)
+		return TEST_ERROR;
+	l2->h_proto = bpf_htons(ETH_P_IPV6);
```

Comment:
oh, did you need that? It should be handled by the pkt builder, when it iterates over the layers during `pktgen__finish()`.

Reply:
It is handled by the pkt builder. I had some problems when initially developing the test and added these statements while something else was my culprit. Consider this removed.